packages/web-api/src/WebClient.ts

```diff
@@ -388,6 +394,170 @@ export class WebClient extends Methods {
     })();
   }
 
+  /* eslint-disable no-trailing-spaces */
+  /**
+   * Handles steps required to upload single or multiple files to Slack.
+   * This is a replacement for the (now legacy) files.upload method of uploading
+   * a single file.
+   * 
+   * **#1**: For each file submitted with this method, submit filenames
+   * and file metadata to files.getUploadURLExternal to request a URL to
+   * which to send the file data to and an id for the file
+   * 
+   * **#2**: for each returned file upload URL, upload corresponding file
+   * 
+   * **#3**: For each successful upload, completeUploadExternal endpoint,
+   * with a list of completed - channel id, initial comment, thread_ts
+   * @param options
+   */
+  public async filesUploadV2(options: FilesUploadV2Arguments = {}): Promise<WebAPICallResult> {
+    // 1
+    const fileUploads = await this.getAllFileUploads(options);
+    const fileUploadsURLRes = await this.fetchAllUploadURLExternal(fileUploads);
+    fileUploadsURLRes.forEach((res, idx) => {
+      const { status } = res;
+      if (status === 'rejected') {
+        this.logger.error(`Error initiating upload for ${fileUploads[idx].filename}`);
```

Comment:
@seratch - Hmm... I think that made an assumption that each file_upload was independent, and so my initial instinct was that users would find this useful, because having all 10 uploads fail because one upload is incorrect seemed annoying. But then I do see that this pattern is supported in your implementation: 

```python
# New way with multiple files!
response = client.files_upload_v2(
    file_uploads=[
        {
            "file": "./logo.png",
            "title": "New company logo",
        },
        {
            "content": "Minutes ....",
            "filename": "team-meeting-minutes-2022-03-01.md",
            "title": "Team meeting minutes (2022-03-01)",
        },
    ],
    channel="C12345",
    initial_comment="Here is the latest version of our new company logo :wave:",
)
```
Which sounds like the scenario where you'd want the entire upload to fail IF there are multiple uploads associated with a single initial_comment / message. Currently my changes don't account for this.

Reply:
@srajiang I think that developers would like to know any of the files are not properly uploaded because it's a partial failure. Also, I haven't checked this yet but when any of the file IDs that are passed to the complete API call (= sharing the files) fails, the sharing process may not work as you expect.

Reply:
Okay - it's true that it's simpler to fail the moment there's a single failure. 

I do still need to add some logic to support a single initial_comment / message + multiple file uploads at the same time so I will work on those both tomorrow.

Reply:
@seratch Now we are immediately throwing an exception when there's a failure in any step, and have added changes to support the multiple file uploads pattern above.